exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/managed/PriorityQueueCopierWrapper.java

```diff
@@ -345,6 +348,33 @@ public void close() {
     @Override
     public SelectionVector4 getSv4() { return null; }
 
+    @Override
+    public void updateOutputContainer(VectorContainer container, SelectionVector4 sv4,
+                                      RecordBatch.IterOutcome outcome, BatchSchema schema) {
+      if (outcome == EMIT) {
+        throw new UnsupportedOperationException("It looks like Sort is hitting memory pressure and forced to spill " +
+          "for cases with EMIT outcome. This Sort is most likely used within the subquery between Lateral and Unnest " +
+          "in which case spilling is unexpected.");
+      }
+
+      VectorContainer dataContainer = getContainer();
+      // First output batch, let's create the container.
+      if (container.getNumberOfColumns() == 0) {
+        for (VectorWrapper<?> vw : dataContainer) {
+          container.add(vw.getValueVector());
+        }
+        // In future when we want to support spilling with EMIT outcome then we have to create SV4 container all the
+        // time. But that will have effect of copying data again by SelectionVectorRemover from SV4 to SV_None. Other
+        // than that we have to send OK_NEW_SCHEMA each time. There can be other operators like StreamAgg in downstream
+        // as well, so we cannot have special handlnig in SVRemover for EMIT phase.
```

Comment:
(1) Typo: "handling"
(2) The usage of the External Sort between Unnest and Lateral is very limited (maybe only if under a Streaming-Aggr), thus the multiple OK_NEW_SCHEMA sounds like a good solution (and StreamAgg needs to be updated to match).

Reply:
Fixed (1). For (2) it can also be used just for OrderBy ? The issue with producing OK_NEW_SCHEMA with each batch is that we don't handle this case properly in all the operators and hence there can be pre-mature failures of the query.